File: memory-domain/src/main/java/com/memory/dto/relationship/response/RelationshipResponse.java
package com.memory.dto.relationship.response;

import com.memory.domain.relationship.Relationship;
import com.memory.domain.relationship.RelationshipStatus;

import java.time.LocalDateTime;

public record RelationshipResponse(
        Long id,
        Long memberId,
        Long relatedMemberId,
        RelationshipStatus relationshipStatus,
        LocalDateTime startDate,
        LocalDateTime endDate
) {

    public static RelationshipResponse from(Relationship relationship) {
        return new RelationshipResponse(
                relationship.getId(),
                relationship.getMember().getId(),
                relationship.getRelatedMember().getId(),
                relationship.getRelationshipStatus(),
                relationship.getStartDate(),
                relationship.getEndDate()
        );
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/relationship/response/RelationshipResponse.kt
package com.memory.dto.relationship.response

import com.memory.domain.relationship.Relationship
import com.memory.domain.relationship.RelationshipStatus
import com.memory.dto.member.response.MemberResponse
import com.memory.dto.member.response.MemberResponse.Companion.from
import java.time.LocalDateTime

data class RelationshipResponse(
    val id: Long?,
    val member: MemberResponse?,
    val relatedMember: MemberResponse?,
    val relationshipStatus: RelationshipStatus?,
    val startDate: LocalDateTime?,
    val endDate: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(relationship: Relationship): RelationshipResponse {
            return RelationshipResponse(
                relationship.id,
                from(relationship.member),
                from(relationship.relatedMember),
                relationship.relationshipStatus,
                relationship.startDate,
                relationship.endDate
            )
        }
    }
}